izumi-reflect/izumi-reflect/src/main/scala_3/izumi/reflect/TagMacro.scala

```diff
@@ -1,71 +1,68 @@
 package izumi.reflect
 
-import scala.quoted.*
+import scala.quoted.{Expr, Quotes, Type}
 
 import izumi.reflect.macrortti.LightTypeTag
-import izumi.reflect.dottyreflection.Inspect
+import izumi.reflect.dottyreflection.{Inspect, InspectorBase}
 
 object TagMacro {
-
-  implicit inline def tag[A <: AnyKind]: Tag[A] = ${ createTagExpr[A] }
-
   def createTagExpr[A <: AnyKind: Type](using Quotes): Expr[Tag[A]] =
     new TagMacro().createTagExpr[A]
 }
 
-class TagMacro(using val ctx: Quotes) {
-  import ctx.reflect.*
+final class TagMacro(using override val qctx: Quotes) extends InspectorBase {
+  import qctx.reflect._
+
+  override def shift: Int = 0
 
   def createTagExpr[A <: AnyKind: Type]: Expr[Tag[A]] = {
-    val typeRepr = TypeRepr.of[A].dealias
-    if (allPartsStrong(typeRepr)) '{ Tag(classOf[Any], Inspect.inspect[A]) }
-    else {
-      val result = mkTag(typeRepr)
-      result.asInstanceOf[Expr[Tag[A]]]
+    val typeRepr = TypeRepr.of[A]
+    if (allPartsStrong(typeRepr)) {
+      // NB: this should always work, but currently causes `TestModel$::ApplePaymentProvider is not a type lambda, it cannot be parameterized` test failure
```

Comment:
@kitlangton There's probably a case when an AppliedType is higher-kinded without a wrapping TypeLambda TypeTree, I wonder if we could detect it somehow or perhaps there may already be a method for fixing up such types (re detection: there should probably be a type lambda representing a kind in `.baseTypes` but it may be hard to process just from that)